### PR TITLE
[sailfishsecrets] Start sailfish-secretsd upon boot. Fixes JB#41547

### PR DESCRIPTION
**Pull request by rainemak**

### CODE REVIEW

File: daemon/sailfish-secretsd.privileges [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# Tells invoker to run sailfish-secretsd as nemo:privileged
```
Comment (Denis-Semakin) [inline daemon/sailfish-secretsd.privileges]: What is this file for?
I didn't find it in .spec file. Don't we put it on file system? Is this file included in package?

File: rpm/sailfish-secrets.spec [hunk 2]
```diff
@@ -161,8 +172,19 @@ rm -rf %{buildroot}
...
+mkdir -p %{buildroot}/%{_datadir}/dbus-1/services/
+mkdir -p %{buildroot}/%{_datadir}/mapplauncherd/privileges.d/
+
 cp -R lib/Secrets/doc/html/* %{buildroot}/%{_docdir}/Sailfish/Secrets/
 cp -R lib/Crypto/doc/html/* %{buildroot}/%{_docdir}/Sailfish/Crypto/
+install -m 0644 daemon/%{secretsdaemon}.service %{buildroot}/%{user_unitdir}
+install -m 0644 daemon/%{secretsdaemon}.privileges %{buildroot}/%{_datadir}/mapplauncherd/privileges.d/
+install -m 0644 daemon/org.sailfishos.secrets.daemon.discovery.service %{buildroot}/%{_datadir}/dbus-1/services/
+
+mkdir -p %{buildroot}/%{_datadir}/mapplauncherd/privileges.d/
```
Comment (monich) [inline rpm/sailfish-secrets.spec]: Copy/paste artifact? It's already done a few lines earler

Comment (rainemak) [reply rpm/sailfish-secrets.spec]: Thanks, fixed

File: rpm/sailfish-secrets.spec [hunk 3]
```diff
@@ -265,3 +291,14 @@ cp -R lib/Crypto/doc/html/* %{buildroot}/%{_docdir}/Sailfish/Crypto/
 %postun -n libsailfishcryptoplugin
 /sbin/ldconfig || :
 
+%post -n %{secretsdaemon}
+systemctl-user daemon-reload || :
+systemctl-user reload-or-try-restart %{secretsdaemon} || :
```
Comment (jpetrell) [inline rpm/sailfish-secrets.spec]: Aren't these things problematic during OS updating secrets daemon is probably not even running?

Comment (rainemak) [reply rpm/sailfish-secrets.spec]: Note that service file has "Conflicts=shutdown.target jolla-actdead-charging.service"

**Comment by rainemak** [comment]
Fixed and merged all linking problems => this works now. Feel free to merge.

**Comment by blammit** [comment]
LGTM. Tested with latest update from this morning; the service is in user-session.target.wants and sailfish-secretsd launches on boot with invoker as nemo as expected.

**Pull request merged**